src/main/vcpf4/usbd_cdc_vcp.c

```diff
@@ -165,10 +163,15 @@ uint32_t CDC_Send_DATA(uint8_t *ptrBuffer, uint8_t sendLength)
     return sendLength;
 }
 
+uint32_t CDC_Send_FreeBytes(void)
+{
+	/* return the bytes free in the circular buffer */
+    return ((APP_Rx_ptr_out - APP_Rx_ptr_in) + (-((int)(APP_Rx_ptr_out <= APP_Rx_ptr_in)) & APP_RX_DATA_SIZE)) - 1;
```

Comment:
Performance is important here as this is evaluated on every byte moved into the buffer. This method requires no evaluation step, it has to be more performant.

Readability is important, but not at the expense of slowing the CPU pipeline down. I'll make a comment to aid in readability.